src/components/checkboxFilter/index.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+
+import { chakra } from '@chakra-ui/react';
+
+import Stack from '../stack';
+import Checkbox from '../checkbox';
+
+type Item<ItemKey> = {
+  key: ItemKey;
+  label: string;
+};
+
+type Props<ItemKey extends string> = {
+  /**
+   * Name of the filter - only used on the DOM.
+   */
+  name: string;
+  /**
+   * @template ItemKey
+   * @param Item: { key: ItemKey, label: string }
+   */
+  items: Item<ItemKey>[];
+  /**
+   * The checkbox filter is a controlled component and the updated filter value must be passed in order to see the correct state.
+   */
+  checked: { [key in ItemKey]: boolean };
+  /**
+   * Numeric value shown next to the checkbox label. Indicates how many search results are going to be visible after the checkbox has been applied.
+   */
+  facets: { [key in ItemKey]: number };
+  /**
+   * Callback function that is triggered after any checkbox has been clicked.
+   * @param item: contains the modified checkbox with the new value
+   * @param newFilterState: contains the new state of the whole filter group
+   */
+  onApply: (
+    item: { key: ItemKey; isChecked: boolean },
+    newFilterState: { [key in ItemKey]: boolean }
+  ) => void;
+};
+
+const addThousandSeparatorToNumber = (value: number) => {
+  return new Intl.NumberFormat('de-CH').format(value);
+};
+
+function CheckboxFilter<ItemKey extends string>({
```

Comment:
I see there is Chakra `CheckboxGroup` and `useCheckboxGroup` custom hook. Not sure is it possible and worth to use it for our case.